OpenRA.Game/FileSystem/InstallShieldCABExtractor.cs

```diff
@@ -0,0 +1,440 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using ICSharpCode.SharpZipLib.Zip.Compression;
+using ICSharpCode.SharpZipLib.Zip.Compression.Streams;
+namespace OpenRA.FileSystem
+{
+	public class InstallShieldCABExtractor : IDisposable, IFolder
+	{
+		const uint FILESPLIT = 0x1;
+		const uint FILEOBFUSCATED = 0x2;
+		const uint FILECOMPRESSED = 0x4;
+		const uint FILEINVALID = 0x8;
+
+		const uint LINKPREV = 0x1;
+		const uint LINKNEXT = 0x2;
+		const uint MAXFILEGROUPCOUNT = 71;
+
+		struct FileGroup
+		{
+			public readonly string Name;
+			public readonly uint FirstFile;
+			public readonly uint LastFile;
+			public FileGroup(BinaryReader reader, long offset)
+			{
+				var nameOffset = reader.ReadUInt32();
+				reader.ReadBytes(0x12);
```

Comment:
I always feel like we're missing something and throwing away data when doing this.
Can you please add comments explaining why we don't use those bytes here and in the other places where you read but don't store them.